src/libexpr/meson.build

```diff
@@ -0,0 +1,164 @@
+# Nix lib expr build file
+#============================================================================
+
+
+# dependencies
+#============================================================================
+
+libexpr_dep_list = [
+    boost_dep,
+    libdl_dep,
+    libsodium_dep,
+    gc_dep,
+
+    libutil_dep,
+    libstore_dep,
+    libmain_dep,
+    libfetchers_dep]
+
+
+# src files
+#============================================================================
+
+libexpr_src = files(
+    'attr-path.cc',
+    'attr-set.cc',
+    'common-eval-args.cc',
+    'eval-cache.cc',
+    'eval.cc',
+    'function-trace.cc',
+    'get-drvs.cc',
+    'json-to-value.cc',
+    'nixexpr.cc',
+    'primops.cc',
+    'value-to-json.cc',
+    'value-to-xml.cc')
+
+libexpr_headers = files(
+    'attr-path.hh',
+    'attr-set.hh',
+    'common-eval-args.hh',
+    'eval-cache.hh',
+    'eval.hh',
+    'eval-inline.hh',
+    'function-trace.hh',
+    'get-drvs.hh',
+    'json-to-value.hh',
+    'nixexpr.hh',
+    'primops.hh',
+    'symbol-table.hh',
+    'value.hh',
+    'value-to-json.hh',
+    'value-to-xml.hh')
+
+
+# include directories
+#========================================================================
+
+# include current dir
+#---------------------------------------------------
+libexpr_inc = [include_directories('.')]
+
+# add subdirectories
+#---------------------------------------------------
+libexpr_dirs = [
+  'flake',
+  'primops']
+
+foreach dir : libexpr_dirs
+    subdir(dir)
+endforeach
+
+
+# targets
+#============================================================================
+
+libexpr_src += custom_target(
+    'parser_tab.[cchh]',
+    output : [
+        'parser-tab.cc',
+        'parser-tab.hh'],
+    input : 'parser.y',
+    command : [
+        bison,
+        '-v',
+        '--output=@OUTPUT0@',
+        '--defines=@OUTPUT1@',
+        '@INPUT@'])
+
+libexpr_src += custom_target(
+    'lexer_tab.[cchh]',
+    output : ['lexer-tab.cc', 'lexer-tab.hh'],
+    input : 'lexer.l',
+    command : [
+        flex,
+        '--outfile=@OUTPUT0@',
+        '--header-file=@OUTPUT1@',
+        '@INPUT@'])
+
+libexpr_src += custom_target(
+  'imported-drv-to-derivation.nix.gen.hh',
+  output : 'imported-drv-to-derivation.nix.gen.hh',
+  input : 'imported-drv-to-derivation.nix',
+  command : [bash, '-c', gen_rheader, 'sh', '@OUTPUT@'])
+
+
+libexpr_src += custom_target(
```

Comment:
Wonder if you could use a generator for these build targets that are all so similar.

Reply:
i think this is a new feature? https://mesonbuild.com/Reference-manual_functions.html#generator

Reply:
> only if these are private headers, generators are bad for headers because they're re-invoked each time they're used. They're really good for things like "custom DSL -> .c" conversion.

yeah meson guidelines usually recommend against using generators